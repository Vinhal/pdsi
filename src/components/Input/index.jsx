import React from 'react'
import TextField from '@material-ui/core/TextField'

const Input = ({ name, label, value, onChange, ...props }) => (
    <TextField
        id={name}
        label={label}
        value={value}
        margin="normal"
        variant="outlined"
        style={{ width: '100%' }}
        InputLabelProps={{ shrink: true }}
        onChange={(ev) => onChange(ev.target.value)}
        {...props}
    />
)

export default Input
