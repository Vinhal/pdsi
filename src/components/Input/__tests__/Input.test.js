import React from 'react'
import { shallow, mount } from 'enzyme'
import Input from '..'

test('Should match Input snapshot', () => {
    const component = shallow(<Input />)

    expect(component).toMatchSnapshot()
})

test('Should change Input value', () => {
    const value = 'valueMock'
    const onChange = jest.fn()

    const wrapper = mount(<Input value='' onChange={onChange} />)

    wrapper.find('input').simulate('change', { target: { value: value }})

    expect(onChange).toHaveBeenCalled()
})

test('Should test Input name', () => {
    const expectedName = 'NAME'

    const wrapper = mount(<Input name={expectedName} />)

    const name = wrapper.props().name

    expect(expectedName).toBe(name)
})
