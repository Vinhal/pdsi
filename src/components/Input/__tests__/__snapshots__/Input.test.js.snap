// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match Input snapshot 1`] = `
<WithStyles(ForwardRef(TextField))
  InputLabelProps={
    Object {
      "shrink": true,
    }
  }
  margin="normal"
  onChange={[Function]}
  style={
    Object {
      "width": "100%",
    }
  }
  variant="outlined"
/>
`;
